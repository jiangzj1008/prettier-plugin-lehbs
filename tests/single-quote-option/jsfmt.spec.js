run_spec(__dirname, { singleQuote: false });
run_spec(__dirname, { singleQuote: true });
