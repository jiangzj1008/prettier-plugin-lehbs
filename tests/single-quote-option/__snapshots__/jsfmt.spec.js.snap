// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`arguments.hbs 1`] = `
<GlimmerComponent
  class="medium"
  @autocomplete="off"
  @errors={{or this.aProp (and this.bProp (get bike "number" "message"))}}
/>

<GlimmerComponent
  class='medium'
  @autocomplete='off'
  @errors={{or this.aProp (and this.bProp (get bike 'number' 'message'))}}
/>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<GlimmerComponent
  class="medium"
  @autocomplete="off"
  @errors={{or this.aProp (and this.bProp (get bike "number" "message"))}}
/>

<GlimmerComponent
  class="medium"
  @autocomplete="off"
  @errors={{or this.aProp (and this.bProp (get bike "number" "message"))}}
/>
`;

exports[`arguments.hbs 2`] = `
<GlimmerComponent
  class="medium"
  @autocomplete="off"
  @errors={{or this.aProp (and this.bProp (get bike "number" "message"))}}
/>

<GlimmerComponent
  class='medium'
  @autocomplete='off'
  @errors={{or this.aProp (and this.bProp (get bike 'number' 'message'))}}
/>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<GlimmerComponent
  class='medium'
  @autocomplete='off'
  @errors={{or this.aProp (and this.bProp (get bike 'number' 'message'))}}
/>

<GlimmerComponent
  class='medium'
  @autocomplete='off'
  @errors={{or this.aProp (and this.bProp (get bike 'number' 'message'))}}
/>
`;

exports[`attributes.hbs 1`] = `
<div title='My title'></div>
<div title="My title"></div>
<div title='My "title"'></div>
<div title="My other 'title'"></div>

<div title="{{t 'my.title'}}" />
<div title='{{t "my.title"}}' />

<a href="/{{url}}/{{url}}"></a>
<a href='/{{url}}/{{url}}'></a>

<button class="padding width-{{locales.length}} {{if (has locale) "uk-button-primary"}}"></button>
<button class="padding width-{{locales.length}} {{if (has locale) 'uk-button-primary'}}"></button>
<button class='padding width-{{locales.length}} {{if (has locale) "uk-button-primary"}}'></button>
<button class='padding width-{{locales.length}} {{if (has locale) 'uk-button-primary'}}'></button>

<div class="padding {{if foo (if fooAgain "bar" (if fooAgainAgain "bar" "foo"))}} baz" />
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<div title="My title"></div>
<div title="My title"></div>
<div title='My "title"'></div>
<div title="My other 'title'"></div>

<div title="{{t 'my.title'}}"></div>
<div title="{{t 'my.title'}}"></div>

<a href="/{{url}}/{{url}}"></a>
<a href="/{{url}}/{{url}}"></a>

<button
  class="padding width-{{locales.length}}
    {{if (has locale) 'uk-button-primary'}}"
></button>
<button
  class="padding width-{{locales.length}}
    {{if (has locale) 'uk-button-primary'}}"
></button>
<button
  class="padding width-{{locales.length}}
    {{if (has locale) 'uk-button-primary'}}"
></button>
<button
  class="padding width-{{locales.length}}
    {{if (has locale) 'uk-button-primary'}}"
></button>

<div
  class="padding
    {{if foo (if fooAgain 'bar' (if fooAgainAgain 'bar' 'foo'))}}
    baz"
></div>
`;

exports[`attributes.hbs 2`] = `
<div title='My title'></div>
<div title="My title"></div>
<div title='My "title"'></div>
<div title="My other 'title'"></div>

<div title="{{t 'my.title'}}" />
<div title='{{t "my.title"}}' />

<a href="/{{url}}/{{url}}"></a>
<a href='/{{url}}/{{url}}'></a>

<button class="padding width-{{locales.length}} {{if (has locale) "uk-button-primary"}}"></button>
<button class="padding width-{{locales.length}} {{if (has locale) 'uk-button-primary'}}"></button>
<button class='padding width-{{locales.length}} {{if (has locale) "uk-button-primary"}}'></button>
<button class='padding width-{{locales.length}} {{if (has locale) 'uk-button-primary'}}'></button>

<div class="padding {{if foo (if fooAgain "bar" (if fooAgainAgain "bar" "foo"))}} baz" />
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<div title='My title'></div>
<div title='My title'></div>
<div title='My "title"'></div>
<div title="My other 'title'"></div>

<div title='{{t "my.title"}}'></div>
<div title='{{t "my.title"}}'></div>

<a href='/{{url}}/{{url}}'></a>
<a href='/{{url}}/{{url}}'></a>

<button
  class='padding width-{{locales.length}}
    {{if (has locale) "uk-button-primary"}}'
></button>
<button
  class='padding width-{{locales.length}}
    {{if (has locale) "uk-button-primary"}}'
></button>
<button
  class='padding width-{{locales.length}}
    {{if (has locale) "uk-button-primary"}}'
></button>
<button
  class='padding width-{{locales.length}}
    {{if (has locale) "uk-button-primary"}}'
></button>

<div
  class='padding
    {{if foo (if fooAgain "bar" (if fooAgainAgain "bar" "foo"))}}
    baz'
></div>
`;

exports[`string-literals.hbs 1`] = `
{{"abc"}}
{{'abc'}}
{{" \\" \\" ' more double quote than single quote "}}
{{' \\' \\' " more single quote than double quote '}}
{{' " \\' \\" \\\\ '}}
{{" \\" \\' ' \\\\ "}}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{"abc"}}
{{"abc"}}
{{' " " \\' more double quote than single quote '}}
{{" ' ' \\" more single quote than double quote "}}
{{' " \\' \\" \\\\ '}}
{{" \\" \\' ' \\\\ "}}
`;

exports[`string-literals.hbs 2`] = `
{{"abc"}}
{{'abc'}}
{{" \\" \\" ' more double quote than single quote "}}
{{' \\' \\' " more single quote than double quote '}}
{{' " \\' \\" \\\\ '}}
{{" \\" \\' ' \\\\ "}}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{{'abc'}}
{{'abc'}}
{{' " " \\' more double quote than single quote '}}
{{" ' ' \\" more single quote than double quote "}}
{{' " \\' \\" \\\\ '}}
{{" \\" \\' ' \\\\ "}}
`;
